src/nni_manager/training_service/local/localTrainingServiceForGPU.ts

```diff
@@ -78,7 +78,7 @@ class LocalTrainingServiceForGPU extends LocalTrainingService {
     }
 
     protected onTrialJobStatusChanged(trialJob: LocalTrialJobDetailForGPU, oldStatus: TrialJobStatus): void {
-        if (trialJob.gpuIndices !== undefined && trialJob.gpuIndices.length !== 0) {
+        if (trialJob.gpuIndices !== undefined && trialJob.gpuIndices.length !== 0 && this.gpuScheduler !== undefined) {
             if (oldStatus === 'RUNNING' && trialJob.status !== 'RUNNING') {
                 for (const index of trialJob.gpuIndices) {
                     this.availableGPUIndices[index] = false;
```

Comment:
why assigning `false`?

Reply:
if the gpu becomes available, it will be assigned `false`?

Reply:
@yds05

Reply:
I'm not very confident about the motivation of assigning false here because I'm not the original author of LocalTrainingService, but I think Yan's change is safe here because if this.gpuScheduler is undefined(means gpuNum is set to 0 in nniconfig file), we should not allocate GPU for trialJob.